### ISSUE TITLE
ImageMagick seems to be very poor in compressing with Quality 100

### ISSUE THREAD

**Issue by RubenKelevra** [libjxl/libjxl#1516]
I noted in #1492 that Gimp and ImageMagick do produce wildly different compression results with quality 100:

I have the suspicion that ImageMagick does not switch to lossless modular when `-quality 100` is set by the user.

Can you have a look at what ImageMagick outputs when just `-quality 100` and a `.jxl` as output is selected? I suspect that there's a bug there somewhere given the large size differences of the files.

_Originally posted by @RubenKelevra in https://github.com/libjxl/libjxl/issues/1492#issuecomment-1152230669_

**Comment by sboukortt** [comment]
You are right that there appears to be a discrepancy. In the files that you have attached, the (much larger) ImageMagick file uses modular mode while the GIMP file seems to use VarDCT.

```
$ jxlinfo IMG_20220603_144349_0_draw_gimp.jxl 
JPEG XL file format container (ISO/IEC 18181-2)
JPEG XL image, 4000x3000, lossy, 16-bit Grayscale
Color space: Grayscale, D65, sRGB transfer function, rendering intent: Relative

$ jxlinfo IMG_20220603_144349_0_draw_imagemagick.jxl 
JPEG XL image, 4000x3000, (possibly) lossless, 16-bit Grayscale
Color space: Grayscale, D65, sRGB transfer function, rendering intent: Relative
```

**Comment by RubenKelevra** [comment]
@sboukortt Ah, thanks for the tool hint and the analysis! :) So gimp seems to make it wrong then. I'll make sure thats the case and report it there! :)
